edit/edit.js

```diff
@@ -0,0 +1,2068 @@
+    debounce(autocompleteOnTyping, 100, cm, info, true);
+    return;
+  }
+  if (info.text.last.match(/[-\w!]+$/)) {
+    cm.state.autocompletePicked = false;
+    cm.options.hintOptions.completeSingle = false;
+    cm.execCommand('autocomplete');
+    setTimeout(() => {
+      cm.options.hintOptions.completeSingle = true;
+    });
+  }
+}
+
+function autocompletePicked(cm) {
+  cm.state.autocompletePicked = true;
+}
+
+function refocusMinidialog(cm) {
+  const section = cm.getSection();
+  if (!section.querySelector('.CodeMirror-dialog')) {
+    return;
+  }
+  // close the currently opened minidialog
+  cm.focus();
+  // make sure to focus the input in newly opened minidialog
+  setTimeout(function() {
+    section.querySelector('.CodeMirror-dialog').focus();
+  }, 0);
+}
+
+function nextPrevEditor(cm, direction) {
+  cm = editors[(editors.indexOf(cm) + direction + editors.length) % editors.length];
+  makeSectionVisible(cm);
+  cm.focus();
+}
+
+function getEditorInSight(nearbyElement) {
+  // priority: 1. associated CM for applies-to element 2. last active if visible 3. first visible
+  let cm;
+  if (nearbyElement && nearbyElement.className.indexOf('applies-') >= 0) {
+    cm = getSectionForChild(nearbyElement).CodeMirror;
+  } else {
+    cm = editors.lastActive;
+  }
+  if (!cm || offscreenDistance(cm) > 0) {
+    const sorted = editors
+      .map(function(cm, index) { return {cm: cm, distance: offscreenDistance(cm), index: index}; })
+      .sort(function(a, b) { return a.distance - b.distance || a.index - b.index; });
+    cm = sorted[0].cm;
+    if (sorted[0].distance > 0) {
+      makeSectionVisible(cm);
+    }
+  }
+  return cm;
+
+  function offscreenDistance(cm) {
+    const LINES_VISIBLE = 2; // closest editor should have at least # lines visible
+    const bounds = cm.getSection().getBoundingClientRect();
+    if (bounds.top < 0) {
+      return -bounds.top;
+    } else if (bounds.top < window.innerHeight - cm.defaultTextHeight() * LINES_VISIBLE) {
+      return 0;
+    } else {
+      return bounds.top - bounds.height;
+    }
+  }
+}
+
+function updateLintReport(cm, delay) {
+  if (delay == 0) {
+    // immediately show pending csslint messages in onbeforeunload and save
+    update(cm);
+    return;
+  }
+  if (delay > 0) {
+    setTimeout(cm => { cm.performLint(); update(cm); }, delay, cm);
+    return;
+  }
+  const state = cm.state.lint;
+  if (!state) {
+    return;
+  }
+  // user is editing right now: postpone updating the report for the new issues (default: 500ms lint + 4500ms)
+  // or update it as soon as possible (default: 500ms lint + 100ms) in case an existing issue was just fixed
+  clearTimeout(state.reportTimeout);
+  state.reportTimeout = setTimeout(update, state.options.delay + 100, cm);
+  state.postponeNewIssues = delay == undefined || delay === null;
+
+  function update(cm) {
+    const scope = cm ? [cm] : editors;
+    let changed = false;
+    let fixedOldIssues = false;
+    let state;
```

Comment:
That's what I ended up using 😸 ... oddly, I had to switch the outer `state` definition to use ` var`

```js
var state = cm.state.lint;
```

otherwise there were JS errors inside the `update` function stating it was undefined. Maybe it's too early in the morning, but I can't wrap my head around that one.